internal/api/api_client.go

```diff
@@ -1,22 +1,68 @@
-package nozzle
+package api
 
 import (
-	"fmt"
 	"net/url"
 	"strings"
+	"sync"
 	"time"
 
-	metrics "github.com/rcrowley/go-metrics"
-
 	cfclient "github.com/cloudfoundry-community/go-cfclient"
 	cache "github.com/patrickmn/go-cache"
+	metrics "github.com/rcrowley/go-metrics"
+	"github.com/wavefronthq/cloud-foundry-nozzle-go/internal/config"
+	"github.com/wavefronthq/cloud-foundry-nozzle-go/internal/utils"
+	"github.com/wavefronthq/go-metrics-wavefront/reporting"
+)
+
+var (
+	internalTags    = utils.GetInternalTags()
+	appCacheExp     = 6 * time.Hour
+	appCache        = cache.New(appCacheExp, time.Hour)
+	appCacheErrors  = utils.NewCounter("cache.errors", internalTags)
+	appCacheMiss    = utils.NewCounter("cache.miss", internalTags)
+	appCacheChannel = make(chan string, 1000)
+	appCacheAPI     *APIClient
+	appCacheDoOnce  sync.Once
 )
 
+func init() {
+	reporting.RegisterMetric("cache.size", metrics.NewFunctionalGauge(func() int64 { return int64(appCache.ItemCount()) }), internalTags)
+	go func() {
```

Comment:
Seems a bit odd to be doing this in an init function given that the cache requires the API client created within the `NewAPIClient` method. 

It'd be better to refactor this into a separate `cache.go` file that provides a `newCache(cfclient, channel)` constructor to create a cache. And make the cache an internal field of the APIClient. Let the cache constructor initiate the go routine to listen on the relevant channel and process the requests.